File: web-client/src/types/post.ts
import { BackgroundKey } from "@/constants";
import { EFileType, EPostType, EPrivacy, EReactionType, TopReact, User } from ".";

export type Post = {
    id: string;
    content: string;
    medias: Media[];
    createdAt: string; // format: dd/MM/yyyy HH:mm:ss
    updatedAt: string; // format: dd/MM/yyyy HH:mm:ss
    privacy: EPrivacy;
    type: EPostType;
    user: User;
    noShared: number;
    noComments: number;
    background: BackgroundKey;
    noReactions: number;
    topReacts: TopReact[];
    reacted?: EReactionType;
    sharedPost: Post;
    groupInfo: GroupInfo;
    taggedUsers: User[];
    mine: boolean;
    saved: boolean;
    illegal: boolean;
};

export type ShareInfo = {
    id: string;
    status: string;
    sharedAt: string; // format: dd/MM/yyyy HH:mm:ss
    sharedUser: User;
    privacy: EPrivacy;
};

export type GroupInfo = {
    id: string;
    name: string;
    avatar: string;
    private: boolean;
};

export type Media = {
    id?: string;
    url: string;
    type: EFileType;
    ownerId: string;
    thumbnail: string;
}

File: server/search-service/src/main/java/vn/edu/tdtu/model/data/Post.java
package vn.edu.tdtu.model.data;

import co.elastic.clients.elasticsearch.license.PostResponse;
import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import vn.edu.tdtu.dto.TopReacts;
import vn.edu.tdtu.enums.EPostType;
import vn.edu.tdtu.enums.EPrivacy;
import vn.edu.tdtu.enums.EReactionType;

import java.io.Serializable;
import java.util.Date;
import java.util.List;

@AllArgsConstructor
@NoArgsConstructor
@Data
@JsonIgnoreProperties(ignoreUnknown = true)
public class Post implements Serializable {
    private String id;
    private String content;
    @JsonFormat(pattern = "dd/MM/yyyy HH:mm:ss")
    private Date createdAt;
    @JsonFormat(pattern = "dd/MM/yyyy HH:mm:ss")
    private Date updatedAt;
    private EPrivacy privacy;
    private EPostType type;
    private User user;
    private int noShared;
    private long noComments;
    private int noReactions;
    private List<TopReacts> topReacts;
    private EReactionType reacted;
    private String background;
    private PostResponse sharedPost;
    private GroupInfo groupInfo;
    private List<Media> medias;
    private List<User> taggedUsers;
    private boolean isMine;
    private boolean saved;
    private boolean illegal;
}
